core/extensions/impl/crypto_extension.cpp

```diff
@@ -41,6 +42,21 @@ namespace kagome::extensions {
     memory_->storeBuffer(out_ptr, common::Buffer(hash));
   }
 
+  void CryptoExtension::ext_keccak_256(runtime::WasmPointer data,
+                                       runtime::SizeType len,
+                                       runtime::WasmPointer out_ptr) {
+    const auto &buf = memory_->loadN(data, len);
+    std::array<uint8_t, 32> hash{};
+    sha3_HashBuffer(256,
```

Comment:
we duplicate code here. It is better to add `std::shared_ptr<Hasher> hasher_` field to the crypto extension and invoke keccak_256 method from it. If hasher has methods for blake and twox, then please update ext_blake and ext_twox methods as well